### PR TITLE
mayhem integration

### PR DESCRIPTION
**Pull request by evanrichter**

### CODE REVIEW

File: aquatic_udp/src/workers/request/mod.rs [hunk 1]
```diff
@@ -15,7 +15,7 @@ use aquatic_udp_protocol::*;
 use crate::common::*;
```
Comment (xansec) [inline aquatic_udp/src/workers/request/mod.rs]: Were these functions made public to support fuzzing?

File: aquatic_udp/src/workers/socket/storage.rs [hunk 2]
```diff
@@ -118,104 +118,109 @@ impl PendingScrapeResponseSlab {
     }
```
Comment (xansec) [inline aquatic_udp/src/workers/socket/storage.rs:118]: This is a pretty significant change, is there a way these changes can be made in a different file?

**Review by xansec — CHANGES_REQUESTED** [review]
It looks like a lot of files were modified to support fuzzing. Just double checking that these changes were necessary (adding arbitrary attr is fine, other changes are not as intuitive). Please make the requested changes.

File: aquatic_udp/src/workers/socket/storage.rs (hunk 2 above)
Comment (evanrichter) [reply aquatic_udp/src/workers/socket/storage.rs:118]: I managed to make the diff smaller here, but the effect is the same: harness the quickcheck test code as if it were a fuzz harness. I've found that quickcheck only runs a couple dozen iterations per run, and sometimes fuzzing it for longer turns up results.

File: aquatic_udp/src/workers/request/mod.rs (hunk 1 above)
Comment (evanrichter) [reply aquatic_udp/src/workers/request/mod.rs]: yes, some functions and types as well. I wanted to fuzz incoming UDP packets without IO, but also without modifying much of their code. Exporting types was the least intrusive way to do that.

That said, my impression of this project is that it is a binary, and the library crates are not realistically intended for external use. This is a common strategy to speed up compile times for projects with lots of code. So we can see what upstream thinks, but they could be totally fine with it. If not, I could put the harness code (in fuzz_targets) in the library crate exposed only under a `#[cfg(fuzzing)]`

**Comment by xansec** [comment]
OK, LGTM!

**Pull request merged**